cmd/occollector/app/collector/processors.go

```diff
@@ -310,17 +312,32 @@ func startProcessor(v *viper.Viper, logger *zap.Logger) (consumer.TraceConsumer,
 	}
 
 	// Wraps processors in a single one to be connected to all enabled receivers.
-	if multiProcessorCfg.Global != nil && multiProcessorCfg.Global.Attributes != nil {
+	hasAttributesProcessing := multiProcessorCfg.Global != nil &&
+		multiProcessorCfg.Global.Attributes != nil &&
+		(len(multiProcessorCfg.Global.Attributes.Values) > 0 ||
+			len(multiProcessorCfg.Global.Attributes.KeyReplacements) > 0)
+	if hasAttributesProcessing {
```

Comment:
Should we maybe leave this inital check just to check `multiProcessorCfg.Global != nil && multiProcessorCfg.Global.Attributes != nil` ?  We check the other two conditions again below anyway.

Reply:
there is a possible way to make both Values and Replacements to be empty see https://github.com/census-instrumentation/opencensus-service/pull/536#discussion_r281872489 for details

Reply:
We would chain the building:
Eg (in psuedo code ish)
```
tp := multiconsumer.NewTraceProcessor(traceConsumers)
if multiProcessorCfg.Global != nil && multiProcessorCfg.Global.Attributes != nil {
    if doAdd {
        tp = addattributesprocessor.NewTraceProcessor(tp, ....)
    }
    if doRemap {
        tp = attributekeyprocessor.NewTraceProcessor(tp, ...) 
    }
}
return tp
```